nameable.rb

```diff
@@ -0,0 +1,39 @@
+require_relative 'person'
+
+class Nameable
+  def correct_name
+    raise NotImplementedError, 'not implemented'
+  end
+end
+
+class BaseDecorator < Nameable
+  atrr_accessor :nameable
+
+  def initilize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitilizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.capitilize
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name[0, 10]
+  end
+end
+
```

Comment:
- [ OPTIONAL ] I highly recommend putting each class in sperate  file to make your project well structured since you are making inheritance on the classes so kindly remove each class to seprate files ✅🔥💯💯